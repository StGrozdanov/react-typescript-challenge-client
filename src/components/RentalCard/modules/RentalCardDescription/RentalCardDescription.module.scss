@use 'src/scss/common/animations' as *;

.description {
    padding: 0px 1rem;
    font-weight: bold;
    @include smoothTransitionAnimation();
}