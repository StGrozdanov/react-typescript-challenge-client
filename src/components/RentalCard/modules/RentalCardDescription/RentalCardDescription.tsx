import { RentalDescriptionProps } from "../../types/interfaces";
import styles from "./RentalCardDescription.module.scss"

const RentalCardDescription: React.FC<RentalDescriptionProps> = ({ rentalName }: RentalDescriptionProps) => {
    return (
        <p data-testid="rentalCardDescription" className={styles.description}>
            {rentalName}
        </p>
    )
}

export default RentalCardDescription;