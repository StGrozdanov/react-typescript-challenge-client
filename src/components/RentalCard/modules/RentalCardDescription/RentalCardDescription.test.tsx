import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import RentalCardDescription from './RentalCardDescription';

describe('RentalCardDescription Component', () => {
  const rentalName = 'Sample Rental Name';

  beforeEach(() => {
    render(<RentalCardDescription rentalName={rentalName} />);
  });

  test('renders the rental description', () => {
    const descriptionElement = screen.getByTestId('rentalCardDescription');

    expect(descriptionElement).toBeInTheDocument();
    expect(descriptionElement).toHaveTextContent(rentalName);
  });

  test('matches the snapshot', () => {
    const { container } = render(<RentalCardDescription rentalName={rentalName} />);
    expect(container).toMatchSnapshot();
  });
});