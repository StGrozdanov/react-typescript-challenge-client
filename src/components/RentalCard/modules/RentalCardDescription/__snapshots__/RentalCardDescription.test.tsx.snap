// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RentalCardDescription Component matches the snapshot 1`] = `
<div>
  <p
    class="description"
    data-testid="rentalCardDescription"
  >
    Sample Rental Name
  </p>
</div>
`;
